include/circt/Dialect/SV/SVExpressions.td

```diff
@@ -11,11 +11,28 @@
 //===----------------------------------------------------------------------===//
 
 include "mlir/Interfaces/InferTypeOpInterface.td"
+include "SVTypes.td"
 
 def HWValueOrInOutType : AnyTypeOf<[HWValueType, InOutType]>;
 def HasCustomSSAName : DeclareOpInterfaceMethods<OpAsmOpInterface,
                          ["getAsmResultNames"]>;
 
+
+def SystemFunctionOp : SVOp<"system"> {
+  let summary = "Simple SV System Function calls";
+  let description = [{
+    This operation calls the indicated system verilog system function.  This
+    supports functions which take normal expression arguments.
+
+    See section 20 of the 2012 SV spec.
+  }];
+  let arguments = (ins StrAttr:$fnName, Variadic<AnyType>:$args);
+  let results = (outs AnyType:$out);
+
+  let assemblyFormat = "$fnName `(` $args `)` attr-dict `:` functional-type($args, $out)";
+}
```

Comment:
Having a catch-all op seems weird to me from a modeling perspective. I admit that this is expedient (and there are a ton of system functions and tasks). However, we already have some of these modeled as explicit ops, e.g., `FinishOp`.

I think we will eventually need more explicit modeling of most of this, even if it's just syntactic modeling. E.g., is a catch-all op going to be problematic for writing verifiers?

WDYT?

Reply:
I suspect eventually we will have explicit ops for all standard functions.  But we only mostly gain type checking for most of those.  For lots of them, we don't care about what they do, so having a generic seems fine.